src/lib.rs

```diff
@@ -21,6 +21,111 @@ use std::env::var;
+    /// Get implementation of the Numeric locale category.
+    fn get_numeric(&mut self) -> Option<Box<Numeric>> { None }
+
+    /// Get implementation of the Time locale category.
+    fn get_time(&mut self) -> Option<Box<Time>> { None }
+}
+
+/// Auxiliary class for creating composing partial implementations of locale factories.
+// FIXME: Create (doc) test when there actually is another implementation to substitute.
+#[derive(Debug, Clone)]
+pub struct CompositeLocaleFactory<First: LocaleFactory, Second: LocaleFactory> {
+    first: First,
+    second: Second,
+}
+
+impl<F: LocaleFactory, S: LocaleFactory> CompositeLocaleFactory<F, S> {
+    pub fn new(first: F, second: S) -> Self {
+        CompositeLocaleFactory::<F, S> {
+            first: first, second: second
+        }
+    }
+}
+
+impl<F: LocaleFactory, S: LocaleFactory> LocaleFactory for CompositeLocaleFactory<F, S> {
+    // XXX: Make a macro for this
+    fn get_numeric(&mut self) -> Option<Box<Numeric>> {
+        if let Some(v) = self.first.get_numeric() {
+            Some(v)
+        } else {
+            self.second.get_numeric()
+        }
+    }
+
+    fn get_time(&mut self) -> Option<Box<Time>> {
+        if let Some(v) = self.first.get_time() {
+            Some(v)
+        } else {
+            self.second.get_time()
+        }
+    }
+}
+
+/// Factory of invariant locales.
+///
+/// Invariant locale, called "C" or "POSIX" by standard C library locale functions, is default
+/// locale definitions for when no information about desired locale is available or localization is
+/// turned off.
+#[derive(Debug, Clone, Default)]
+pub struct InvariantLocaleFactory;
+
+impl InvariantLocaleFactory {
+    /// Constructs invariant locale factory.
+    ///
+    /// The signature is just so that it matches the other locale factories so the classes can be
+    /// substituted depending on target operating system and the code using them does not have to
+    /// care.
+    #[allow(unused_variables)]
+    pub fn new(locale: &str) -> Self {
+        InvariantLocaleFactory
+    }
+}
+
+impl LocaleFactory for InvariantLocaleFactory {
+    // NOTE: Yep, it's empty. This just returns nothing and the Locale constructor will take care
+    // of the actual defaults.
+}
+
+#[cfg(target_os = "linux")]
+pub mod linux;
+
+#[cfg(target_os = "linux")]
+pub use linux::LibCLocaleFactory as SystemLocaleFactory;
+
+#[cfg(not(target_os = "linux"))]
+pub use InvariantLocaleFactory as SystemLocaleFactory;
+
+/// Return LocaleFactory appropriate for default user locale, as far as it can be determined.
+///
+/// The returned locale factory provides locale facets implemented using standard localization
+/// functionality of the underlying operating system and configured for user's default locale.
+pub fn user_locale_factory() -> SystemLocaleFactory {
+    // FIXME: Error handling? Constructing locale with "" should never fail as far as I can tell.
+    SystemLocaleFactory::new("").unwrap()
```

Comment:
Ah, thanks. Obviously I need to connect my fork to Travis and I need to compile for Windows (I've set myself a cross toolchain), so I catch inconsistency between the system-specific implementations.